.github/workflows/static-analysis.yml

```diff
@@ -0,0 +1,29 @@
+name: Static Analysis
+
+on:
+  push:
+    branches: [master]
+  pull_request:
+    branches: [master]
+
+jobs:
+  pre-commit:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Install micromamba
+        uses: mamba-org/provision-with-micromamba@v13
+        with:
+          environment-file: environment-dev.yml
+          environment-name: xtensor-dev
+          cache-downloads: true
+      - name: Set PYTHON_HASH
+        shell: bash -l {0}
+        run: echo "PYTHON_HASH=$(python -VV | sha256sum | cut -d' ' -f1)" >> $GITHUB_ENV
+      - uses: actions/cache@v3
+        with:
+          path: ~/.cache/pre-commit
+          key: precommit-v1-${{ env.PYTHON_HASH }}-${{ hashFiles('.pre-commit-config.yaml') }}
+      - name: Pre-commit hooks
+        shell: bash -l {0}
+        run: python -m pre_commit run --all-files --show-diff-on-failure
```

Comment:
Is the goal just to run pre-commit? For me this looks much simpler:
```
name: pre-commit

on:
  pull_request:
  push:
    branches: [main]

jobs:
  pre-commit:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v3
    - uses: actions/setup-python@v2
    - uses: pre-commit/action@v2.0.3
 ```
But it is true that this just checks, and does not push

Reply:
Yes, this is just running precommit and caching. I took the step from the [doc](https://pre-commit.com/#github-actions-example), but I missed the pointer to the official action.